tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/BUILD

```diff
@@ -12,6 +12,36 @@ java_binary(
     runtime_deps = [":lcov_merger_lib"],
 )
 
+genrule(
+    name = "rename_build_tools",
+    srcs = ["BUILD.tools"],
+    outs = ["coverage/BUILD"],
+    cmd = "cp $< $@"
+)
+
+genrule(
+    name = "create_empty_workspace",
+    outs = ["WORKSPACE"],
+    cmd = "touch $@"
+)
+
+genrule(
+    name = "coverage_output_generator_zip",
+    srcs = [
+        ":all_lcov_merger_tools_deploy.jar",
+        "coverage/BUILD",
+        "WORKSPACE",
+    ],
+    outs = ["coverage_output_generator.zip"],
+    cmd = "zip -q -j $@ $(SRCS)",
```

Comment:
nit: Why not just do the cp and touch in this genrule so that the silly rules above are not necessary?

Reply:
The advantage of having these trivial rules is explicitly defining and controlling the inputs and outputs. Doing everything in one genrule leads to more maintenance burdens...

Reply:
k